Gems/LyShine/Code/Source/LyShinePass.cpp

```diff
@@ -0,0 +1,283 @@
+#include <Atom/RPI.Public/Pass/PassAttachment.h>
+#include <Atom/RPI.Public/Image/ImageSystemInterface.h>
+#include <Atom/RPI.Public/Image/AttachmentImagePool.h>
+#include <Atom/RPI.Public/Scene.h>
+#include <Atom/RPI.Reflect/Pass/RasterPassData.h>
+#include <AzCore/std/iterator.h>
+
+#include "LyShinePass.h"
+
+namespace LyShine
+{
+    AZ::RPI::Ptr<LyShinePass> LyShinePass::Create(const AZ::RPI::PassDescriptor& descriptor)
+    {
+        return aznew LyShinePass(descriptor);
+    }
+
+    LyShinePass::LyShinePass(const AZ::RPI::PassDescriptor& descriptor)
+        : Base(descriptor)
+    {
+    }
+
+    LyShinePass::~LyShinePass()
+    {
+        LyShinePassRequestBus::Handler::BusDisconnect();
+    }
+
+    void LyShinePass::ResetInternal()
+    {
+        LyShinePassRequestBus::Handler::BusDisconnect();
+
+        Base::ResetInternal();
+    }
+
+    void LyShinePass::BuildInternal()
+    {
+        AZ::RPI::Scene* scene = GetScene();
+        if (scene)
+        {
+            // Listen for rebuild requests
+            LyShinePassRequestBus::Handler::BusConnect(scene->GetId());
+
+            RemoveChildren();
+
+            // Get the current list of render targets being used across all loaded UI Canvases
+            LyShine::AttachmentImagesAndDependencies attachmentImagesAndDependencies;
+            LyShinePassDataRequestBus::EventResult(
+                attachmentImagesAndDependencies,
+                scene->GetId(),
+                &LyShinePassDataRequestBus::Events::GetRenderTargets
+            );
+
+            AddRttChildPasses(attachmentImagesAndDependencies);
+            AddUiCanvasChildPass(attachmentImagesAndDependencies);
+        }
+
+        Base::BuildInternal();
+    }
+
+    void LyShinePass::RebuildRttChildren()
+    {
+        QueueForBuildAndInitialization();
+    }
+
+    AZ::RPI::RasterPass* LyShinePass::GetRttPass(const AZStd::string& name)
+    {
+        for (auto child:m_children)
+        {
+            if (child->GetName() == AZ::Name(name))
+            {
+                return azrtti_cast<AZ::RPI::RasterPass*>(child.get());
+            }
+        }
+        return nullptr;
+    }
+
+    AZ::RPI::RasterPass* LyShinePass::GetUiCanvasPass()
+    {
+        return m_uiCanvasChildPass.get();
+    }
+
+    void LyShinePass::AddRttChildPasses(LyShine::AttachmentImagesAndDependencies attachmentImagesAndDependencies)
+    {
+        for (const auto& attachmentImageAndDependencies : attachmentImagesAndDependencies)
+        {
+            AddRttChildPass(attachmentImageAndDependencies.first, attachmentImageAndDependencies.second);
+        }
+    }
+
+    void LyShinePass::AddRttChildPass(AZ::Data::Instance<AZ::RPI::AttachmentImage> attachmentImage, AttachmentImages attachmentImageDependencies)
+    {
+        // Add a pass that renders to the specified texture
+
+        // Create a pass template
+        auto passTemplate = AZStd::make_shared<AZ::RPI::PassTemplate>();
```

Comment:
Option: It seems you can make a RttChildPasss.pass asset file for this pass template since everything are hard-coded except pass data. I think pass data can be provided when create pass. 
It's up to you. 
With asset file you may add/remote slots and connections more easily without changing the code.